internal/framework/fwtest/schema_validation.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Splunk, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fwtest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"go.uber.org/multierr"
+)
+
+func ResourceSchemaValidate(res resource.Resource, model any) (errs error) {
+	var resp resource.SchemaResponse
+	res.Schema(context.TODO(), resource.SchemaRequest{}, &resp)
+
+	if resp.Schema.Description == "" {
+		errs = multierr.Append(errs, errors.New("missing schema description"))
+	}
+
+	if len(resp.Schema.Attributes) == 0 {
+		return multierr.Combine(errs, errors.New("missing schema attribute definitions"))
+	}
+
+	v := reflect.ValueOf(model)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	if v.Kind() != reflect.Struct {
+		return multierr.Combine(errs, fmt.Errorf("model must be a struct, provided: %T", model))
+	}
+
+	expected := make(map[string]attr.Value)
+	for i := range v.Type().NumField() {
+		var (
+			field = v.Type().Field(i)
+			named = field.Tag.Get("tfsdk")
+		)
+		if named == "" {
+			continue
+		}
+		if !field.IsExported() {
+			errs = multierr.Append(errs, fmt.Errorf("framework requires field exported %q", field.Name))
+			continue
+		}
+		// TODO(MovieStoreGuy): This is fine until we handle more complex types.
+		switch f := reflect.Zero(field.Type).Interface().(type) {
+		case attr.Value:
+			expected[named] = f
+		default:
+			// Handle the case where the field is potentially nested or complex.
+			panic("You have entered the twilight zone. Please check the model definition.")
+		}
```

Comment:
I purposely ignored this since I have no reason to make it more complicated at the moment, but I imagine that will come once nested models are adopted.